megaavr/cores/dxcore/wiring_analog.c

```diff
@@ -917,14 +921,67 @@ void analogReadSampleNum(uint8_t numSamples) {
       ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC128_gc;
       break;
     default:
-        badArg("Unexpected value");
+      badArg("Unexpected value");
       break;
     }
-  }
 }
 
 void analogReadEnableSingleEnded() {
   while (ADC0.COMMAND & ADC_STCONV_bm);
 
   ADC0.CTRLA &= ~ADC_CONVMODE_bm;
 }
+
+//Assuming default of 2, can ask bob about this
+void analogReadPrescaler(uint8_t prescaler) {
+  while (ADC0.COMMAND & ADC_STCONV_bm);
+
+  ADC0.CTRLC &= 0b11110000;   //Setting prescaler to 2
```

Comment:
I'm thinking it would be better to validate the input before this. That way the register won't get cleared if the input is invalid. Or you could save the register value and restore it in the default case below if the input is not valid.